util/devel/mason-completion.bash

```diff
@@ -0,0 +1,275 @@
+  cur=${COMP_WORDS[COMP_CWORD]}
+  case "$cur" in 
+    -*) 
+      COMPREPLY=($(compgen -W "$optional $optional_abbr $update_tag" -- "${COMP_WORDS[COMP_CWORD]}" ))
+      ;;
+     *)
+      ;;
+  esac
+}
+
+_complete_mason_add(){
+  _options add
+}
+
+_complete_mason_init(){
+  local cur
+  cur=${COMP_WORDS[COMP_CWORD]}
+  case "$cur" in
+    -*)
+      _options init 
+      ;;
+    *)
+      COMPREPLY=($(compgen -d -S / -- "$cur"))
+      ;;
+  esac
+}
+
+_complete_mason_new(){
+  _options new
+}
+_complete_mason_option(){
+  local optional
+  local optional_abbr
+  optional=$(mason --help | grep -v '^[A-Za-z]' | grep '^.*--' |      \
+            sed -e 's;^\(.*--[^ ]*\).*$;\1;' | sed -e 's;-[A-Za-z],;   ;' | \
+            xargs echo)
+  optional_abbr=$(mason --help | grep -v '^[A-Za-z]' | grep '^.*--' |        \
+          sed -e 's;^\(.*--[^ ]*\).*$;\1;' | grep ',' | sed -e 's;,.*$;;' | \
+          xargs echo)
+  COMPREPLY=($(compgen -W "$optional $optional_abbr" -- "$1" ))
+}
+
+_complete_mason_command(){
+  local cur
+  local prev
+  local arg
+  cur=${COMP_WORDS[COMP_CWORD]}
+  prev="${COMP_WORDS[COMP_CWORD-1]}"
+  arg=$(mason --help | grep -v '^[A-Za-z]' | grep -v '^.*--'  | \
+     grep -v '<command>' | grep -v '\[options\]'  | sed 's/^[ \t]*//' | \
+     cut -d" " -f1 | xargs echo)
+    COMPREPLY=($(compgen -W "$arg" "${COMP_WORDS[1]}"))
+}
+
+_mason(){
+  local cur
+  COMPREPLY=()
+  cur=${COMP_WORDS[COMP_CWORD]}
+  if [[ ${COMP_CWORD} -lt 2 ]]
+  then
+    case "$cur" in
+      -*)
+        _complete_mason_option "$cur"
+        ;;
+      *)
+        _complete_mason_command "$cur"
+    esac
+  fi
+  if [[ ${COMP_CWORD} -ge 2 ]]
+  then
+    local cur=${COMP_WORDS[COMP_CWORD]}
+    local prev=${COMP_WORDS[COMP_CWORD-1]}
+    local alt_prev=${COMP_WORDS[COMP_CWORD-2]}
+    local subcmd
+    local arg
+    arg=$(mason --help | grep -v '^[A-Za-z]' | grep -v '^.*--'  | \
+       grep -v '<command>' | grep -v '\[options\]'  | sed 's/^[ \t]*//' | \
+       cut -d" " -f1 | xargs echo)
+
+    for i in $arg
+    do
+      if [[ $prev == "$i" ]]
+      then
+        subcmd=$prev
+        break;
+      elif [[ $alt_prev == "$i" ]]
+      then
+        subcmd=$alt_prev
+        break;
+      fi
+    done
+    case "$subcmd" in 
+      'new')
+        _complete_mason_new "$cur"
+        ;;
+      'init')
+        _complete_mason_init "$cur"
+        ;;
+      'add')
+        _complete_mason_add "$cur"
+        ;;
+      'rm')
+        _complete_mason_add "$cur"
+        ;;
+      'run')
+        _complete_mason_run "$cur"
+        ;;
+      'build')
+        _complete_mason_build "$cur"
+        ;;
+      'update')
+        _complete_mason_update "$cur"
+        ;;
+      'search')
+        _complete_mason_search "$cur"
+        ;;
+      'env')
+        _complete_mason_env "$cur"
+        ;;
+      'clean')
+        _complete_mason_clean "$cur"
+        ;;
+      'doc')
+        _complete_mason_doc "$cur"
+        ;;
+      'test')
```

Comment:
This is showing up as `est` for some reason:

```
> mason <tab-tab>
add       clean     env       external  new       rm        search    update
build     doc       est       init      publish   run       system
```

Reply:
Could you share a screenshot ? Its showing correct output on my machine 
![adw](https://user-images.githubusercontent.com/40923324/89541319-d9eb1400-d81b-11ea-9b90-b197484e96fe.PNG)

Reply:
Maybe it's an MacOS thing... I'll try to debug a bit. Let me know if you have any ideas.

![Screen Shot 2020-08-06 at 10 20 32 AM](https://user-images.githubusercontent.com/3595449/89543080-7e466980-d7ce-11ea-9941-62a423ca32a5.png)

Reply:
- Try typing `mason test <tab>` and check if the completion works correctly. 
- Type `mason -h` and check if `test` is displayed correctly there, since the completions are parsed from there.

Reply:
Thanks. I think I identified the cause and have a suggested work-around [here](https://github.com/chapel-lang/chapel/pull/16035#discussion_r466454960)